mlir/test/Target/LLVMIR/Import/instructions.ll

```diff
@@ -370,6 +370,19 @@ define void @load_store(ptr %ptr) {
 
 ; // -----
 
+; CHECK-LABEL: @invariant_load
+; CHECK-SAME:  %[[PTR:[a-zA-Z0-9]+]]
+define float @invariant_load(ptr %ptr) {
+  ; CHECK:  %[[V:[0-9]+]] = llvm.load %[[PTR]] {alignment = 4 : i64, invariantLoad} : !llvm.ptr -> f32
```

Comment:
Can you add the invariant flag to the assembly format since it is an inherent attribute. I know we don't do this consistently but at least for new flags it would be nice to not print them in the attribute dictionary.

I would probably add it between atomic ordering and attr-dict:
```
    (`volatile` $volatile_^)? $addr
    (`atomic` (`syncscope` `(` $syncscope^ `)`)? $ordering^)?
    (`invariant` $isInvariant^)? 
    attr-dict `:` qualified(type($addr)) `->` type($res)
```
After that a small roundtrip test may make sense as well.

Reply:
Good point, will do